File: apps/ruby/routes/request/validate-query.rb
require 'primate/route'
require 'primate/pema'

schema = Pema.schema({
  'baz' => Pema.int,
  'foo' => Pema.string
})

Route.get do |request|
  begin
    parsed = request.query.parse(schema, true)
    parsed
  rescue Pema::ValidationError => e
    e.message
  end
end

File: apps/python/routes/request/validate-query.py
from primate import Route, pema

schema = pema.schema({"baz": pema.int(), "foo": pema.string()})


@Route.get
def handle_get(request):
    try:
        parsed = request.query.parse(schema, True)
        return parsed
    except pema.ValidationError as e:
        return str(e)
